pkg/kubefed2/util/yaml_writer.go

```diff
@@ -0,0 +1,45 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"github.com/ghodss/yaml"
+	"io"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+func WriteUnstructuredToYaml(unstructuredObj *unstructured.Unstructured, w io.Writer) error {
```

Comment:
I think it's potentially surprising that a reusable utility method intended to output yaml modifies its input.  That was fine when the code was internal to `enable`, but now that it's available for reuse I think either documentation or a DeepCopy of the input before modification is suggested.

Reply:
Updated!